backend-lib/src/main/rust/lib.rs

```diff
@@ -236,7 +236,7 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initDataD
         let mut db_data = wallet_db(env, network, db_data)
             .map_err(|e| anyhow!("Error while opening data DB: {}", e))?;
 
-        let seed = (!seed.is_null()).then(|| SecretVec::new(env.convert_byte_array(seed).unwrap()));
+        let seed = utils::java_nullable_bytes_to_rust(env, &seed)?.map(SecretVec::new);
```

Comment:
Nit: There are two uses of this pattern. It's marginal, but I would add a nullable counterpart to `parse_secret`.